include/particles/particle.tcc

```diff
@@ -108,17 +112,18 @@ bool mpm::Particle<Tdim>::initialise_particle(
     const std::shared_ptr<mpm::Material<Tdim>>& material) {
   bool status = this->initialise_particle(particle);
   if (material != nullptr) {
-    if (this->material_id_ == material->id() ||
-        this->material_id_ == std::numeric_limits<unsigned>::max()) {
+    if (this->material_id() == material->id() ||
```

Comment:
As mentioned in the description, this is still using one-phase with `hdf5` for multiphase developed later right?

Reply:
Yes. The two-phase version of hdf5 will be implemented later. This PR is just to enhance the input material